internal/nginx/config/upstreams_template.go

```diff
@@ -1,11 +1,11 @@
 package config
 
-// FIXME(kate-osborn): Add upstream zone size for each upstream.
-// This should be dynamically calculated based on the number of upstreams.
+// FIXME(kate-osborn): Dynamically calculate upstream zone size based on the number of upstreams.
```

Comment:
would it make sense to document why 512k?
```suggestion
// FIXME(kate-osborn): Dynamically calculate upstream zone size based on the number of upstreams.
// 512k will support up to 648 upstream servers. 
```